content/docs/08.architecture.md

```diff
@@ -67,6 +67,30 @@ You can scale Workers as necessary and have many instances on multiple servers,
 
 As the worker executes the tasks and the polling triggers, it needs access to any external services they use (database, REST services, message broker, etc.). Workers are the only server components that need access to external services.
 
+#### Worker Group (EE)
+
+::alert{type="warning"}
+This is an [Enterprise Edition](https://kestra.io/enterprise) feature.
+::
+
+A Worker Group is a set of workers that can be targeted specifically for a task execution or a polling trigger evaluation. For this, the task or the polling trigger must define the `workerGroup.key` property with the key of the worker group to target. A default worker group can also be configured at the namespace level.
+
+All tasks and polling triggers are executed on the un-named default worker group unless the `workerGroup.key` property is defined.
+
+Worker Groups can be used, for example, for the following needs:
+- Execute tasks and polling triggers on specific computation capabilities (like GPU).
+- Execute tasks and polling triggers on specific OS.
+- Restrict backend access to a set of workers (firewall rules, private networks, ...).
+- Execute tasks and polling triggers close to a remote backend (region selection).
+
+::alert{type="warning"}
+Even if using worker groups, we strongly advise always having at least one worker in the un-named default worker group, as there is no way to enforce that all tasks and polling triggers have a `workerGroup.key` property set.
+::
+
+There is currently no validation that a worker group exists, so a task or a polling trigger that defines a worker group that didn't exist will wait forever pending execution.
```

Comment:
```suggestion
There is currently no way to validate whether a worker group exists before using it in a task or trigger. If a task or a polling trigger defines a worker group that doesn't exist, it will wait forever, leaving the flow's Execution stuck in a pending state.
```